java/daikon/chicory/DeclReader.java

```diff
@@ -296,22 +219,34 @@ public void read(File pathname) throws IOException {
   }
 
   /**
-   * Reads a single declaration from decl_file. The opening "DECLARE" line should have already been
-   * read. Returns the ppt.
+   * Reads a single program point declaration from decl_file.
+   *
+   * @param decl_file EntryReader for reading data
+   * @return DeclPpt for the program point
+   * @throws IOException if there is trouble reading the file
    */
   protected DeclPpt read_decl(EntryReader decl_file) throws IOException {
 
+    String firstLine = decl_file.readLine();
     // Read the name of the program point
-    String pptname = decl_file.readLine();
-    if (pptname == null) throw new Error("file " + decl_file.getFileName() + " ends prematurely");
+    String pptname = firstLine.substring(4); // skip "ppt "
     assert pptname.contains(":::");
     DeclPpt ppt = new DeclPpt(pptname);
     ppts.put(pptname, ppt);
 
+    // Skip the ppt-type record.
```

Comment:
What about `flags` and `parent` as described at http://plse.cs.washington.edu/daikon/download/doc/developer.html#Program-point-declarations ?

Reply:
Many of the record types are not needed when all we care about is comparability values.  I have updated a couple of the comments to reflect this.